lib/src/main/java/com/androidessence/materialdesignspecs/BaseColorAdapter.java

```diff
@@ -0,0 +1,118 @@
+package com.androidessence.materialdesignspecs;
+
+import android.content.Context;
+import android.support.annotation.LayoutRes;
+import android.support.v4.content.ContextCompat;
+import android.support.v7.widget.RecyclerView;
+import android.view.LayoutInflater;
+import android.view.View;
+import android.view.ViewGroup;
+import android.widget.ImageView;
+
+import com.androidessence.materialdesignspeclibrary.R;
+
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * Created by ankitagrawal on 5/15/17.
+ */
+@SuppressWarnings("WeakerAccess")
+public abstract class BaseColorAdapter extends RecyclerView.Adapter<BaseColorAdapter.BaseColorViewHolder> {
+    private List<Integer> colors = new ArrayList<>();
+    private int selectedPosition = -1;
+
+    private ColorDialog.OnColorSelectedListener listener;
+
+    public BaseColorAdapter(List<Integer> colors, ColorDialog.OnColorSelectedListener listener) {
+        this(colors, listener, -1);
+    }
+
+    public BaseColorAdapter(ColorDialog.OnColorSelectedListener listener) {
+        this(new ArrayList<Integer>(), listener, -1);
+    }
+
+    public BaseColorAdapter(List<Integer> colors, ColorDialog.OnColorSelectedListener listener, int selectedPosition) {
+        this.colors = colors;
+        this.listener = listener;
+        this.selectedPosition = selectedPosition;
+    }
+
+    @Override
+    public BaseColorViewHolder onCreateViewHolder(ViewGroup parent, int viewType) {
+        Context context = parent.getContext();
+        View view = LayoutInflater.from(context).inflate(getRVItemLayout(), parent, false);
+        return getViewHolder(view);
+    }
+
+    @Override
+    public void onBindViewHolder(BaseColorViewHolder holder, int position) {
+        holder.bindColor(colors.get(position));
+    }
+
+    public void setSelected(int pos) {
+        selectedPosition = pos;
+        notifyDataSetChanged();
+    }
+
+    public int getSelected() {
```

Comment:
I think we should rename this method to `getSelectedPosition()`. It will be more readable where it's used.